forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java

```diff
@@ -1,116 +1,185 @@
-            FMLHandshakeHandler.LOGGER.debug(FMLHandshakeHandler.FMLHSMARKER, "Clearing");
-
-            // Preserve
-            ServerData serverData = Minecraft.getInstance().getCurrentServer();
-
-            // Clear
-            if (Minecraft.getInstance().level == null) {
-                // Ensure the GameData is reverted in case the client is reset during the handshake.
-                GameData.revertToFrozen();
-            }
-
-            // Clear
-            Minecraft.getInstance().clearLevel(new DirtMessageScreen(new TranslationTextComponent("connect.negotiating")));
-
-            // Restore
-            Minecraft.getInstance().setCurrentServer(serverData);
-        });
-
-        FMLHandshakeHandler.LOGGER.debug(FMLHandshakeHandler.FMLHSMARKER, "Waiting for clear to complete");
-        try {
-            future.get();
-            FMLHandshakeHandler.LOGGER.debug("Clear complete, continuing reset");
-            return true;
-        } catch (Exception ex) {
-            FMLHandshakeHandler.LOGGER.error(FMLHandshakeHandler.FMLHSMARKER, "Failed to clear, closing connection", ex);
-            context.getNetworkManager().disconnect(new StringTextComponent("Failed to clear, closing connection"));
-            return false;
-        }
-    }
+	public static final Field handshakeField;
+	public static final Constructor contextConstructor;
+	static final Logger logger = LogManager.getLogger();
+	static final Marker RESETMARKER = MarkerManager.getMarker("RESETPACKET").setParents(MarkerManager.getMarker("FMLNETWORK"));
+
+	public static SimpleChannel handshakeChannel;
+
+	public ClientReset() {
+		IEventBus bus = FMLJavaModLoadingContext.get().getModEventBus();
+		bus.addListener(ClientReset::init);
+		ModLoadingContext.get().registerExtensionPoint(DisplayTest.class, () -> new DisplayTest(() -> NetworkConstants.IGNORESERVERONLY, (a, b) -> true));
+	}
+
+	@SubscribeEvent
+	public static void init(FMLCommonSetupEvent event) {
+		if (handshakeField == null) {
+			logger.error(RESETMARKER, "Failed to find FML's handshake channel. Disabling mod.");
+			return;
+		}
+		if (contextConstructor == null) {
+			logger.error(RESETMARKER, "Failed to find FML's network event context constructor. Disabling mod.");
+			return;
+		}
+		try {
+			//handshakeField.setAccessible(true);
+			//contextConstructor.setAccessible(true);
+			Object handshake = handshakeField.get(null);
+			if (handshake instanceof SimpleChannel) {
+				handshakeChannel = (SimpleChannel)handshake;
+				logger.info(RESETMARKER, "Registering forge reset packet.");
+				handshakeChannel.messageBuilder(S2CReset.class, 98)
+						.loginIndex(S2CReset::getLoginIndex, S2CReset::setLoginIndex)
+						.decoder(S2CReset::decode)
+						.encoder(S2CReset::encode)
+						.consumerNetworkThread(HandshakeHandler.biConsumerFor(ClientReset::handleReset))
+						.add();
+				logger.info(RESETMARKER, "Registered forge reset packet successfully.");
+			}
+		}
+		catch (Exception e) {
```

Comment:
What is this actually catching? The catch should be more specific and refactored out so it doesn't cover code it doesn't need to